openmls/src/group/public_group/validation.rs

```diff
@@ -515,7 +516,42 @@ impl PublicGroup {
 
     /// Returns a [`LeafNodeValidationError`] if an [`ExtensionType`]
     /// in `extensions` is not supported by a leaf in this tree.
-    #[cfg(test)]
+    pub(crate) fn validate_group_context_extensions_proposal(
+        &self,
+        proposal_queue: &ProposalQueue,
+    ) -> Result<(), GroupContextExtensionsProposalValidationError> {
+        let mut iter = proposal_queue.filtered_by_type(ProposalType::GroupContextExtensions);
+
+        match iter.next() {
+            Some(queued_proposal) => match queued_proposal.proposal() {
+                Proposal::GroupContextExtensions(extensions) => {
+                    let ext_type_list: Vec<_> = extensions
+                        .extensions()
+                        .iter()
+                        .map(|ext| ext.extension_type())
+                        .collect();
+
+                    self.check_extension_support(&ext_type_list).map_err(|_| GroupContextExtensionsProposalValidationError::ExtensionNotSupportedByAllMembers)?
+                }
+                _ => {
+                    return Err(GroupContextExtensionsProposalValidationError::LibraryError(
```

Comment:
I think you can get rid of this library error by using an `if let` construction here.

Reply:
Hm, I could use `if let` but then what if I do find a non-gce proposal in there? I could ignore it or return a LibraryError -- it would just live in the `else` branch of the `if let`.

Reply:
The policy we agreed on at some point was that we ignore libraryerrors where possible. For example in `apply_proposals`, we filter for proposals quite a lot and always ignore "unexpected" results. Do I remember this correctly @franziskuskiefer? We should really document this somewhere if we haven't already.

Reply:
There's a tradeoff between readability, soundness, and correctness. If soundness and correctness isn't impacted, it's ok to ommit library errors. There were some refactorings that unfortunately introduced soundness issues by removing library errors (something that still needs fixing).
Since this function has to return an error anyway because it's checking for multiple gce proposals, I think we should keep the library error here.

Reply:
Fair enough. Have we written all of this down somewhere? If not, maybe the wiki would be a good place? Or the documentation of `LibraryError`?